### PR TITLE
setup: ban FC type in eslintrc

### LINKED ISSUE #411: setup: add an eslint rule to ban the React.FC
### Description and goal

To harmonize the code base, we ban the React.FC

### Acceptance criteria

No React.FC in the code base anymore

### PR DESCRIPTION
**Pull request by Yohh**
- add rules to ban FC, react.FC, react.FunctionComponent & react.FunctionalComponent

close #411

### CODE REVIEW

**Review by SharglutDev — APPROVED** [review]
Lgtm, thank you for this PR.

I know it's a pain but don't you think we should extract the props in a type where there is more than one ?

**Comment by Yohh** [comment]
> Lgtm, thank you for this PR.
> 
> I know it's a pain but don't you think we should extract the props in a type where there is more than one ?

I was wondering myself, it's related to the stories right? We should know why it has been written this way I think.

**Comment by SharglutDev** [comment]
> > Lgtm, thank you for this PR.
> > I know it's a pain but don't you think we should extract the props in a type where there is more than one ?
> 
> I was wondering myself, it's related to the stories right? We should know why it has been written this way I think.

Yep and for the Warped Map as well

File: ui-core/src/components/inputs/datePicker/Calendar.tsx [hunk 1]
```diff
@@ -40,7 +34,7 @@ const Day: React.FC<DayProps> = ({
   </div>
 );
 
-const Calendar: React.FC<CalendarProps> = (props) => {
+const Calendar = (props: CalendarProps) => {
   const { displayedMonthStartDate, onDayClick } = props;
```
Comment (Uriel-Sautron) [inline ui-core/src/components/inputs/datePicker/Calendar.tsx]:
```suggestion
const Calendar = ({ displayedMonthStartDate, onDayClick } : CalendarProps) => {
```

**Review by Uriel-Sautron — APPROVED** [review]
LGTM !! Thanks

File: ui-core/src/components/inputs/datePicker/Calendar.tsx (hunk 1 above)
Comment (Yohh) [reply ui-core/src/components/inputs/datePicker/Calendar.tsx]: done

**Comment by Yohh** [comment]
> > > Lgtm, thank you for this PR.
> > > I know it's a pain but don't you think we should extract the props in a type where there is more than one ?
> > 
> > 
> > I was wondering myself, it's related to the stories right? We should know why it has been written this way I think.
> 
> Yep and for the Warped Map as well

done

**Review by SharglutDev — APPROVED** [review]
Lgtm, thank you

**Pull request merged**